modP2pImpl/src/org/aion/p2p/impl/zero/msg/ResHandshake1.java

```diff
@@ -82,18 +75,34 @@ public static ResHandshake1 decode(final byte[] _bytes) {
         }
     }
 
+    public String getBinaryVersion() {
+        return this.binaryVersion;
+    }
+
     @Override
     public byte[] encode() {
         byte[] superBytes = super.encode();
-        byte[] binaryVersionBytes = this.binaryVersion.getBytes();
+        byte[] binaryVersionBytes = new byte[0];
+        try {
+            binaryVersionBytes = this.binaryVersion.getBytes("UTF-8");
+        } catch (UnsupportedEncodingException e) {
+
+        }
+
         int len = binaryVersionBytes.length;
-        if(len > Byte.MAX_VALUE){
-            binaryVersionBytes = Arrays.copyOfRange(binaryVersionBytes, 0 , Byte.MAX_VALUE - 1);
+        if (len > Byte.MAX_VALUE) {
```

Comment:
To be clear, does this imply that our ``binaryVersion`` fields maximum length is 127 bytes? Not sure if that matches up with the max length we enforce (63 characters), given that UTF-8 can take up to a max of 4 bytes per character:

https://stackoverflow.com/questions/9533258/what-is-the-maximum-number-of-bytes-for-a-utf-8-encoded-character

Reply:
yes, I think @aion-6368726973 did this way - to limit the Max length to 127bytes. Ii will be trunked to 127 if the message length over than 127 bytes

Reply:
binaryVersion maximum length is 127 bytes since that length is encoded as part of the buffer. I do agree about the max length that is enforced. I wasn't too sure on why "63"  was selected as the maximum length. We can change it to match 127 bytes. Thoughts?